### PR TITLE
[ENH] Switch datman's CI config to github actions + update package configuration

### PR DESCRIPTION
**Pull request by DESm1th**
This pull request updates our CI pipelines to use github actions instead of circleci. It also cleans up our existing configuration to remove some clutter. 

- Add github actions workflows that:
  - Automatically label pull requests, and comment on malformed PR titles (this is used later for automated changelogs) (356705900bccdfc30740e7acbcd03bab9f8ae936)
  - Automatically run tests against python 3.8, 3.9 and 3.10 and post test coverage comment on PR (dea9f515072f7ef5ee804a7e7dcf3c40e38a2813)
  - Automatically generate and publish docs to our site (83fb207187e092894b107d9633e07795d15224aa)
  - Automatically generate a release when tags are pushed (generates a changelog, adds changelog to release notes on github, updates zenodo creators list order, and pushes to test-pypi and then pypi) (59c302511a3e5c73a974fc8ed6850a74f7562295)
- Refactor the Dockerfile to speed up builds (74d0d2ffc54d71dc57afd5fe305e6027c4f69e81)
- Move all project configuration into pyproject.toml, and remove setup.py and setup.cfg (b7ea1a0a18db1c5aecaee30027c62ef3e4c70b00, 35faa86506af7ea9a155e45c5fc3edb712e18013, dbd93f15d53609da3ce22c2a18992c67f8439b16)
- Use new importlib.metadata package to version package from git tags, remove versioneer (a27b97da7b05cd1fd33bdc2f992b699887d74e39)
- Fix some docs config/build issues
   - Remove docs/requirements.txt file (is now contained in optional dependencies in pyproject.toml) (41a33f1713deb85c43146ca81bdf59ddf930d67d)
   - Move README.rst to the docs folder (3f0b31bd94fcaacf8075fa42ebea0d1903eeba51)
   - Fix build issue from certain links being declared as having duplicate names (7b373d79dead6f16661142db73e672977997bb82)
   - Remove versioneer references from docs/conf.py (4eb07217ed52d4428f1b6774ae002d1148f95012)
- Update zenodo configuration (8896a94312d43e24ce3f71e1e17181d3189b1f94, 9a0ccb2c00f5d70fe6a0a2683a2e9b81882d1737)
   - Fix format error in .zenodo.json contributors entries
   - Remove .maintenance folder and move update_zenodo.py to .github folder
   - Add some missing creators
   - Modify update_zenodo.py to use ``git-summary --line`` instead of now removed ``git-line-summary`` tool
- Remove old/unused config files, remove versioneer from manifest.in, rename CHANGES.rst to CHANGELOG.md (name change needed by a github action) (5887bb0076dd4a3ee83dd7b39d09634ca772b2fd)
- Tweak auto-assign config to always assign me and randomly choose two others, and to use the github action instead of the old bot (which has sporadically and mysteriously broken on us) (85997ed5291a7418722c2b1b9e5ed69ce48bab2c)
- Adds tests, coverage and zenodo badges to our readme (717df18ee11e02c8a96d6a08bf492621efd03068)
- Updates contributions guidelines, moves the file to our docs, updates link references to it (b6a343b377a048fe6443e48345338da3b44b5472)

### CODE REVIEW

**Comment by DESm1th** [comment]
The 'add a label' and 'tests' check failures are, I believe, because it's trying to use my credentials to run things owned by tigrlab. They should run properly once the PR is merged.

**Comment by DESm1th** [comment]
Oh! Also, @jerdra , @josephmje , @salimansourc do you guys want to be added to the maintainer list? If so what email should I throw on? 

@all-contributors add @josphemje, @jerdra, @salimansourc for code, maintenance, review

**Comment by DESm1th** [comment]
Oh! Also, @jerdra , @josephmje , @salimansourc do you guys want to be added to the maintainer list? If so what email should I throw on? 

@all-contributors add @josephmje , @jerdra, @salimansourc for code, maintenance, review

**Comment by DESm1th** [comment]
Shoot. I saw an issue in their repo that led me to hope you could add multiple users naturally like that. Ha.

@all-contributors add @jerdra for code, maintenance, review

**Comment by DESm1th** [comment]
@all-contributors add @jerdra for code, maintenance, review

**Comment by DESm1th** [comment]
@all-contributors add @salimansourc for code, maintenance, review

**Review by josephmje — DISMISSED** [review]
Thanks for this @DESm1th! Excited about the new Github actions you've added here, this automates so much!

I just left a few very minor comments

File: .github/auto_assign.yml [hunk 1]
```diff
@@ -2,21 +2,25 @@
...
+useReviewGroups: true
+
+# A list of reviewers to be added to pull requests (GitHub user name)
+reviewGroups:
+  alwaysAssign:
+    - DESm1th
+  randomAssign:
+    - jerdra
+    - josephmje
+    - kwitczak
```
[the remark this answers is not in the document]
Comment (DESm1th) [reply .github/auto_assign.yml:20]: Good catch! I was working off the original list we had and for some reason he wasnt on that one.

**Pull request merged**